signing_clients/apps.py

```diff
@@ -182,6 +198,11 @@ def parse(klass, buf):
             # End of section
             if not line:
                 if item:
+                    algos = item.pop('algos')
+                    found_algos = set(item['digests'].keys())
+                    if algos is not None and set(algos) != found_algos:
+                        raise ParsingError("Manifest parsing error: algos "
+                                           "don't match (%d)" % lineno)
```

Comment:
I wonder if we should put `algos` into the error message so that it becomes clearer what's going on and what the mismatch actually is.